docs/Testing.md

```diff
@@ -0,0 +1,132 @@
+# Testing
+
+A test case takes the form of a binary \<basename\>.wasm file, next to that
+that there will always be a \<basename\>.\<ext\> source file from which the
+test was originally compiled from which can be used as a reference in the event
+of an error.
```

Comment:
I like the simplicity of this, but I expect it'll be too limiting as we add more tests in more languages. It'll be convenient to build Rust source files with cargo, and other language source files with their respective build systems, and that'll often require additional support files, so it won't always be as simple as \<basename\>.\<ext\>.

What would you think of a convention where we have a `sources` directory, and in it we have source code and build scripts that one can run to re-generate the wasm files?

Reply:
> What would you think of a convention where we have a sources directory, and in it we have source code and build scripts that one can run to re-generate the wasm files?

Some of the WebAssembly proposals have a `meta` directory that contain wast generators; might want to keep with that convention?

Would still like to see the "final" source files next to the test binaries as part of the build for reference as more often than not assertion errors don't make much sense without looking at the source.

Slight duplication but copying it back out to the parent directory next to the binary as part of the build means the implementor running the tests doesn't have to care how a particular proposal structured the build system.